jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/CorrectThreadCredentialsProvider.kt

```diff
@@ -0,0 +1,33 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.util.ThrowableComputable
+import software.amazon.awssdk.auth.credentials.AwsCredentials
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider
+import software.aws.toolkits.resources.message
+
+/**
+ * Offloads fetching credentials to a background task and a modal progress bar if the current thread is EDT
+ */
+class CorrectThreadCredentialsProvider(private val delegate: AwsCredentialsProvider) :
+    AwsCredentialsProvider by delegate {
```

Comment:
do we need to do this by delegate? The `AwsCredentialsProvider` only has one method and we're overriding it. If other methods were added in the future - we might actually want to hook into them so probably want to get the compilation exception if they were to be added.

Reply:
Ya, will switch todropping delegate